### PR TITLE
Recipe updates for v3.9.0

### PR DESCRIPTION
**Pull request by trexfeathers**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] ~~Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by trexfeathers** [comment]
@conda-forge-admin, please rerender

**Comment by trexfeathers** [comment]
@conda-forge-admin, please rerender

**Comment by trexfeathers** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+{% set version = "3.9.0" %}
 
 package:
   name: iris
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/s/scitools-iris/scitools-iris-{{ version }}.tar.gz
-  sha256: bf1f35053c16c14938ee0a1a6cfa7ab60804cf1ec79bac5b550179354370d4fc
+  url: https://pypi.io/packages/source/s/scitools-iris/scitools_iris-{{ version }}.tar.gz
```
Comment (trexfeathers) [inline recipe/meta.yaml:8]: Note the change in how PyPI now constructs its URLs

**Review by bjlittle — APPROVED** [review]

**Pull request merged**